packages/components/src/higher-order/navigate-regions/index.js

```diff
@@ -37,42 +57,48 @@ export function useNavigateRegions( ref, shortcuts = defaultShortcuts ) {
 		nextRegion.focus();
 		setIsFocusingRegions( true );
 	}
-	const focusPrevious = useCallback( () => focusRegion( -1 ), [] );
-	const focusNext = useCallback( () => focusRegion( 1 ), [] );
 
-	useKeyboardShortcut( shortcuts.previous, focusPrevious, {
-		bindGlobal: true,
-	} );
-	useKeyboardShortcut( shortcuts.next, focusNext, { bindGlobal: true } );
+	const clickRef = useRefEffect(
+		( element ) => {
+			function onClick() {
+				setIsFocusingRegions( false );
+			}
 
-	useEffect( () => {
```

Comment:
We should be using `useRefEffect` here because it will be called when the ref changes.